nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java

```diff
@@ -80,7 +80,7 @@ public abstract class BaseTestIceberg {
 
   private static final Logger LOG = LoggerFactory.getLogger(BaseTestIceberg.class);
 
-  @TempDir public Path temp;
+  @TempDir public File temp;
```

Comment:
what's the reason from switching from `Path` to `File`? The `File` API is considered **old** and `Path` should generally be preferred imo. Also the tests should work without having to change this. 
The only place you'd have to adjust is `TestNessieTable#testCreate()`: `String expected = (temp.toUri() + DB_NAME + "/" + tableName).replace("///", "/")` to `String expected = (temp.toUri() + DB_NAME + "/" + tableName)`

Reply:
> https://github.com/apache/iceberg/pull/4826#discussion_r937079863
I'd rather not mix file system APIs. This uses both Path and File. Can you get rid of Path and make that a string everywhere?

Removed `Path` after this comment.

Reply:
yeah, `Path` is more advanced but we don't need any specific function from `Path` that is not present in `File`.

Reply:
> The only place you'd have to adjust

previously `metadataFiles`, `metadataVersionFiles`, `manifestFiles`, `metadataLocation` used to work with `tableName`, now because of uuid addition, it has to be location based. Hence, we started modifying the test cases and ended up here.

Reply:
I agree that the change here probably isn't necessary. I would probably revert this change

Reply:
> what's the reason for switching from Path to File?

If I switch to `Path` completely,  that also needs many changes as `TestNessieTable` validations/asserts has many  
`Assertions.assertThat(new File(foo)).exists();`
Also, some functions like `metadataFiles` use `listFiles()` that needs to be changed to `Files.walk()`.

Reply:
to me it seems that the switch from `Path` to `File` in this PR isn't required. I actually checked out the branch and reverted that change and everything was working after adjusting `TestNessieTable#testCreate()`

Reply:
Do you mean, just with the **first commit** of this PR?
If different, can you share the branch with me?

Reply:
I mean in this PR. The diff that is being done in this file isn't required imo

Reply:
Oh, you mean just this file. I have reverted now.